File: Sep/Day2/find-the-number-of-ways-to-place-people-i.py
#  Time: O(N ^ 3)  ::  Space: O(1)
class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        count = 0
        n = len(points)

        for i in range(n):
            pointA = points[i]
            for j in range(n):
                pointB = points[j]
                if i == j or not (
                    pointA[0] <= pointB[0] and pointA[1] >= pointB[1]
                ):
                    continue
                if n == 2:
                    count += 1
                    continue

                isPresentInside = False
                for k in range(n):
                    if k == i or k == j:
                        continue

                    pointC = points[k]
                    isXPresentInside= (
                       pointC[0] >= pointA[0] and pointC[0] <= pointB[0]
                    )
                    isYPresentInside = (
                       pointC[1] <= pointA[1] and pointC[1] >= pointB[1]
                    )
                    if isXPresentInside and isYPresentInside:
                        isPresentInside = True
                        break
                if not isPresentInside:
                    count += 1
        return count

File: Sep/Day2/find-the-number-of-ways-to-place-people-i.java
//  Time: O(N ^ 3)  ::  Space: O(1)

import java.util.*;

class Solution {
    public int numberOfPairs(int[][] points) {
        int count = 0;
        int n = points.length;

        for (int i = 0; i < n; i++) {
            int[] pointA = points[i];
            for (int j = 0; j < n; j++) {
                int[] pointB = points[j];
                if (i == j || !(pointA[0] <= pointB[0] && pointA[1] >= pointB[1])) {
                    continue;
                }
                if (n == 2) {
                    count++;
                    continue;
                }

                boolean isPresentInside = false;
                for (int k = 0; k < n; k++) {
                    if (k == i || k == j) continue;
                    int[] pointC = points[k];

                    boolean isXPresentInside = (pointC[0] >= pointA[0] && pointC[0] <= pointB[0]);
                    boolean isYPresentInside = (pointC[1] <= pointA[1] && pointC[1] >= pointB[1]);

                    if (isXPresentInside && isYPresentInside) {
                        isPresentInside = true;
                        break;
                    }
                }
                if (!isPresentInside) {
                    count++;
                }
            }
        }
        return count;
    }
}
